sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -137,6 +137,10 @@ class RocksDB(
   @volatile private var numKeysOnWritingVersion = 0L
   @volatile private var fileManagerMetrics = RocksDBFileManagerMetrics.EMPTY_METRICS
 
+  // SPARK-46249 - Keep track of recorded metrics per version which can be used for querying later
+  // Updates and access to recordedMetrics are protected by the DB instance lock
+  @volatile private var recordedMetrics: Option[RocksDBMetrics] = None
```

Comment:
Let's explicitly add annotation for which lock the field depends on, like below field.

`@GuardedBy("acquireLock")`

Reply:
Done